hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java

```diff
@@ -1258,16 +1291,27 @@ public void updateFeature() {
 		}
 	}
 
-	public RecordStream recordStream() {
-		if (recordStream == null) {
-			recordStream = new RecordStream(
+	public RecordStreamManager recordStreamManager() {
+		return recordStreamManager;
+	}
+
+	/**
+	 * RecordStreamManager should only be initialized after system files have been loaded,
+	 * which means enableRecordStreaming has been read from file
+	 */
+	public void initRecordStreamManager() {
+		try {
+			recordStreamManager = new RecordStreamManager(
 					platform,
 					runningAvgs(),
-					nodeAccount(),
-					properties().getStringProperty("hedera.recordStream.logDir"),
-					properties());
+					PropertiesLoader.isEnableRecordStreaming(),
```

Comment:
Again, no need to change for this PR, but worth noting that `PropertiesLoader` will be removed shortly. Main three reasons are:

1. Methods that directly access `PropertiesLoader` instead of using an injected `PropertySource` are less testable.
2. `PropertiesLoader` puts the default properties in code (`ApplicationConstants`) instead of externalizing (_src/main/resources/bootstrap.properties_). 
3. In `dev` mode, there are race conditions triggered by a `FileUpdate` to `0.0.121` when multiples nodes are running in the same JVM and sharing a classloader-scoped singleton instead of a `ServicesContext`-scoped singleton.

Reply:
got it, thank you very much for the detailed explanation! @tinker-michaelj